torchtext/transforms.py

```diff
@@ -349,6 +359,26 @@ def _tokenize(self, text: str) -> List[str]:
         """
         return self.bpe.tokenize(text)
 
+    def add_special_tokens(self, special_tokens_dict: Dict[str, Union[str, List[str]]]) -> int:
```

Comment:
How did you decide on this implementation? As I remember from the post, the inspiration was primarily from HuggingFace's implementation: https://huggingface.co/docs/transformers/internal/tokenization_utils#transformers.SpecialTokensMixin.add_tokens

Reply:
This implementation is very similar to the HuggingFace's. 

For HuggingFace:
* Mixin defines the special tokens, and the structure enforced on the input: https://github.com/huggingface/transformers/blob/main/src/transformers/tokenization_utils_base.py#L896
* `PreTrainedTokenizer` keeps a separate encoder map for special tokens that are added later: https://github.com/huggingface/transformers/blob/main/src/transformers/tokenization_utils.py#L430 (also creates a trie to save all those tokens that are not to be split: https://github.com/huggingface/transformers/blob/main/src/transformers/tokenization_utils.py#L445)
* Added tokens encoder map is checked first when doing encoding: https://github.com/huggingface/transformers/blob/main/src/transformers/tokenization_utils.py#L586
* Regex based approach is used to distinguish between non-special and special tokens: https://github.com/huggingface/transformers/blob/3f936df66287f557c6528912a9a68d7850913b9b/src/transformers/tokenization_utils.py#L511

Of course, HuggingFace's implementation is more generic and is extensible to all of their tokenizers through inheritance. `GPT2Tokenizer` inherits `PreTrainedTokenizer` which inherits `PreTrainedTokenizerBase`, which implements two mixins: SpecialTokensMixin, PushToHubMixin. Comparatively torchtext's tokenizers all inherit `nn.Module` with their respective (and fairly separated imo) C++ implementation.

This pull request is the minimal implementation of their general Python approach. That's why I kept the Python API to be the same. A more general implementation is possible as C++ supports multiple inheritance. But it might take a bit more time and require a broader discussion with your team.